### ISSUE TITLE
SPM unable to resolve 2.1.1

### ISSUE THREAD

**Issue by amonshiz** [DataDog/dd-sdk-swift-testing#65]
### The issue
We recently bumped our dependency from the following resolved SPM definition
```
{
  "package": "dd-sdk-swift-testing",
  "repositoryURL": "https://github.com/DataDog/dd-sdk-swift-testing",
  "state": {
    "branch": null,
    "revision": "707ba6cebf51ec3bba0697b463450973f0530302",
    "version": "2.1.0"
  }
}
```

to the below
```
{
  "package": "dd-sdk-swift-testing",
  "repositoryURL": "https://github.com/DataDog/dd-sdk-swift-testing",
  "state": {
    "branch": null,
    "revision": "abb4da0970a96686b3c07476d4f4eaae16547c52",
    "version": "2.1.1"
  }
}
```

Now our developers are getting the below error when attempting to resolve packages within Xcode
```
Revision abb4da0970a96686b3c07476d4f4eaae16547c52 for dd-sdk-swift-testing remoteSourceControl https://github.com/DataDog/dd-sdk-swift-testing version 2.1.1 does not match previously recorded value 53f53728eda43f3e39a1066128131ed37d378640
```

#### Dependency Manager:

SPM

#### Xcode version:

Xcode 13.4.1
Xcode 14.0 beta 6

#### macOS version:

macOS Monterey 12.4 (21F79)

**Comment by nachoBonafonte** [comment]
Hi @amonshiz, 
I have re-released with version number 2.1.2, could you try it

**Comment by amonshiz** [comment]
@nachoBonafonte thank you for the fast update! By pointing at 2.1.2 SPM is able to resolve again. Thanks!

**Comment by nachoBonafonte** [comment]
Hi @amonshiz, not related to this but I was wondering if you saw this guide we published recently: [Test configuration](https://docs.datadoghq.com/continuous_integration/guides/test_configurations/).
It has a section at the bottom (Custom configurations), which explains a feature that was developed after the feedback meeting we had with you, it is useful to identify different runtime conditions for the tests (improving the error and flakiness reports)

**Comment by diederich** [comment]
It looks like 2.1.2 just got retagged? 
SPM unfortunately doesn't seem to behave super nicely with changing tags. Once package caches are reset, it fetches the new tag - though now checksums don't seem to match:

```
checksum of downloaded artifact of binary target 'DatadogSDKTesting' (d2031df292638e333bc836194f5a842556216447fddae91311435de9159b3d56) does not match checksum specified by the manifest (088be69d5ec5900806b09ddc850c95ce9dd0984f980e26b857507fd61563af1c)
```

**Comment by nachoBonafonte** [comment]
Yes, it looks so. I also updated the checksum in the Package.swift file, but it looks like SPM is caching the previous checksum. I will release a new version 2.1.3 to solve it.

**Comment by diederich** [comment]
Thank you! 🙏 
Out of curiosity - why the 2.1.2 retag earlier today? 

Also - just saw the test configurations - this looks great! It seems to put the individual configurations of a testplan together. 🥳  
Could it use the name of the configuration in the testplan somewhere, or am I just not seeing it? :-D

**Comment by diederich** [comment]
2.1.3 looks good! 👍 🙏

**Comment by nachoBonafonte** [comment]
> 2.1.3 looks good! 👍 🙏
Nice to hear that.

> Out of curiosity - why the 2.1.2 retag earlier today?

I rebuilt it with Xcode 14.0 to avoid a small issue I noticed when running on iOS 16.0, but it was definitely not the best idea. I should have created a new release instead of changing the tag.

> Could it use the name of the configuration in the testplan somewhere, or am I just not seeing it? :-D
For using it in the test plan set the environment variable DD_TAGS or add a new entry to it, something like:

`DD_TAGS=test.configuration.testplan:yourTestplanName`

Will make the configuration to be called testplan, and will have the value you write there. You will have to put the value by hand in each test plan though

**Comment by diederich** [comment]
> Will make the configuration to be called testplan, and will have the value you write there. You will have to put the value by hand in each test plan though (does it answer you question?, or am I missing something? )

Ohhh - I see. I saw the `DD_TAGS` trick in the docs, but wasn't thinking we could set those in the different testplan configurations. 🙏 That makes total sense - will do now!

**Comment by diederich** [comment]
Is it possible that 2.1.3 got retagged again? 

```
Showing All Issues
Revision 6ccdc2fe1570de1ac1c02f0930532d4ad6218b4f for dd-sdk-swift-testing remoteSourceControl https://github.com/DataDog/dd-sdk-swift-testing version 2.1.3 does not match previously recorded value eb2854e0c2ef51bec865090da65466bde8023aa1
```

**Comment by diederich** [comment]
My hunch is it's this runner / action that reapplies the tag?

<img width="1340" alt="Screenshot 2022-10-13 at 17 47 39" src="https://user-images.githubusercontent.com/61170/195644629-ddc808a8-fc8b-472a-be1b-82fae1097dfa.png">

**Comment by nachoBonafonte** [comment]
> Is it possible that 2.1.3 got retagged again?
No, it was not. I mean I didn't do anything. 

That runner has always reapplied the tag while updates the binary signature of the framework, but it has always worked like that, no changes there.

**Comment by diederich** [comment]
There's no "re-applying" tags IMHO. Public history should be immutable. 

I think here's what happened today:
* the "release" 2.1.3 got created on GitHub
* we updated our Package.resolved, picking up the 2.1.3's SHA (including the 2.1.2 binaries?)
* the release workflow runs
* re-applies the tag to a new commit pointing to the right binaries.

So there's the time window between the tag and the workflow finishing - where users would get broken dependencies?
